### ISSUE TITLE
question about 'gt3d' label

### ISSUE THREAD

**Issue by jay2002** [akanazawa/hmr#51]
Hi @akanazawa, I found the 'gt3d' labels are weird...
1. The 'gt3d' labels in h36m are centralized to zero, which correspond to the augment code 'reflect_joints3d'. But the 'gt3d' labels in mpi_inf_3dhp are not centralized... I guess this will leads incorrect label when performing random_flip. 
2. The 'gt3d' labels in h36m are different to the joints calculated by 'pose' and 'shape' labels in h36m.
3. In addition, the scale and direction of gt3d in mpi_inf_3dhp seems to be different to h36m.

To summarize, I found the all the 'gt3d' labels are not in line with expectations, which will confuse the training. But it seems that the final results are not so bad... Is there any problem with my understanding? Any explanation?

**Comment by akanazawa** [comment]
Hi,

1. The joints are mean centered dynamically as it goes through the data augmentation (and possibly multiple places in the code to make sure). There's always a possibility of a bug, but I visualized these joints after flips so I think they are ok. 

2. If you're referring to SMPL joints, they are different bc gt3d are the 14 subset of H36M joints that comes from the original dataset and different from the SMPL joints. Actually, over here there is a bug, the global pose of the mosh is flipped by 180 degrees when it doesn't have to be, so the gt3d and 3D h36m joints from moshed values are pointing in the opposite direction. 

But this is actually how the model was trained, essentially model just has to ignore the first 3D dimensions of the ground truth 3D pose & figure out it's own. This is what happens & all the other loss instruct the model to ignore the first 3D . I've been meaning to update the models and data, but it is how the model was trained. I actually have trained with the updated data, but there was only a marginal difference in the performance.

3. gt3d of mpi_inf_3dhp and h36m should be similar.. is the orientation different by a lot?

Thanks and sorry for the confusion,

Angjoo

**Comment by jay2002** [comment]
@akanazawa Thx for your reply. I have figured it out.
For 2, I think ignore the first 3 pose coefficients (global pose) in 3d loss computation is more natural. I'll try it on my own data
